src/components/scroll-items-loader.jsx

```diff
@@ -0,0 +1,97 @@
+import React, {
+  useState, useRef, useEffect, cloneElement,
+} from 'react';
+import PropTypes from 'prop-types';
+import v1 from 'uuid';
+
+import '../styles/components/scroll-items-loader.scss';
+
+const ScrollItemsLoader = ({
+  onLoadMoreItems,
+  loadingIndicator,
+  scrollContainer,
+  items,
+  hasMoreItems,
+}) => {
+  const { className: scrollContainerClassName } = scrollContainer.props;
+  const [loading, setLoading] = useState(false);
+  const [loadMoreItems, setLoadMoreItems] = useState(false);
+  const ref = useRef();
+
+  const isNotScrollable = () => {
+    const { scrollHeight, clientHeight } = ref.current;
+    return scrollHeight <= clientHeight;
+  };
+
+  const isBottom = () => {
+    const { scrollHeight, scrollTop, clientHeight } = ref.current;
+    return scrollHeight - Math.ceil(scrollTop) === clientHeight;
+  };
+
+  const checkLoadMoreItems = () => {
+    if (!loading && hasMoreItems && isBottom()) {
+      setLoadMoreItems(true);
+    }
+  };
+
+  useEffect(() => {
+    if (isNotScrollable() && hasMoreItems) {
+      onLoadMoreItems();
+    } else {
+      setLoading(false);
+      setLoadMoreItems(false);
+    }
+  }, [items]);
+
+  useEffect(() => {
+    if (loadMoreItems && hasMoreItems) {
+      setLoading(true);
+      onLoadMoreItems();
+    }
+  }, [loadMoreItems]);
+
+  useEffect(() => {
+    setLoading(false);
+  }, [hasMoreItems]);
+
+  return cloneElement(
+    scrollContainer,
+    {
+      ref,
+      onScroll: checkLoadMoreItems,
+      className: `${scrollContainerClassName || ''} scroll-container`,
+    },
+    [items.length > 0 && items, (loading || items.length === 0) && loadingIndicator],
```

Comment:
This is a bit hard to read, could `[items, loading  && loadingIndicator]` do the same thing?